### ISSUE TITLE
[Bug?]: Yarn scripts not working with PNP

### ISSUE THREAD

**Issue by kegan-myers-wwt** [yarnpkg/berry#3167]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

When running a script such as `yarn test` an error occurs:

```
node:fs:2640
      const dev = BigIntPrototypeToString(stats.dev, 32);
                  ^

TypeError: BigInt.prototype.toString requires that 'this' be a BigInt
    at Number.toString (<anonymous>)
    at gotStat (node:fs:2640:19)
    at /<snip>/.pnp.js:9375:13
```

### To reproduce

```
$ yarn set version berry
$ yarn
$ yarn test
```

package.json
```
{
  "name": "tests",
  "dependencies": {
    "jest": "^27.0.6"
  },
  "scripts": {
    "test": "jest"
  }
}
```

.yarnrc.yml
```
nodeLinker: 'pnp'
```

### Environment

  System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 16.3.0 - /private/var/folders/50/_n2rx9zn1s5dbczb225sm1zh0000gn/T/xfs-5f8b512b/node
    Yarn: 2.4.0 - /private/var/folders/50/_n2rx9zn1s5dbczb225sm1zh0000gn/T/xfs-5f8b512b/yarn
    npm: 7.15.1 - ~/.nvm/versions/node/v16.3.0/bin/npm

### Additional context

It also seems like pnp wasn't used by default for some reason, but that may be some quirk of my local environment. Either way the documentation on yarnpkg.com doesn't indicate that changed behavior anywhere I could find.

**Comment by merceyz** [comment]
Duplicate of https://github.com/yarnpkg/berry/issues/2232

**Comment by merceyz** [comment]
It has been fixed in the 3.0 RC version
```sh
yarn set version 3.0.0-rc.12
```

**Comment by kegan-myers-wwt** [comment]
Is it not going to be backported to `berry`? When is 3.0 going to be stable? It seems weird that the current release version is broken with modern node.

**Comment by kegan-myers-wwt** [comment]
@merceyz this problem is not solved with `3.0.0-rc.12`

```
$ rm -rf .yarn .pnp.js .yarn yarn.lock
$ yarn set version 3.0.0-rc.12
➤ YN0000: Downloading https://github.com/yarnpkg/berry/raw/%40yarnpkg/cli/3.0.0-rc.12/packages/yarnpkg-cli/bin/yarn.js
➤ YN0000: Saving the new release in .yarn/releases/yarn-3.0.0-rc.12.cjs
➤ YN0000: Done in 0s 916ms
$ yarn install
➤ YN0000: ┌ Resolution step
➤ YN0032: │ fsevents@npm:2.3.2: Implicit dependencies on node-gyp are discouraged
➤ YN0032: │ fsevents@npm:2.3.2: Implicit dependencies on node-gyp are discouraged
➤ YN0000: └ Completed in 1s 879ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ xmlchars@npm:2.2.0 can't be found in the cache and will be fetched from the remote re
➤ YN0013: │ y18n@npm:5.0.8 can't be found in the cache and will be fetched from the remote regist
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote reg
➤ YN0013: │ yargs-parser@npm:20.2.9 can't be found in the cache and will be fetched from the remo
➤ YN0013: │ yargs@npm:16.2.0 can't be found in the cache and will be fetched from the remote regi
➤ YN0000: └ Completed in 1s 683ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed in 0s 473ms
➤ YN0000: Done with warnings in 4s 93ms
$ yarn test
node:fs:2640
      const dev = BigIntPrototypeToString(stats.dev, 32);
                  ^

TypeError: BigInt.prototype.toString requires that 'this' be a BigInt
    at Number.toString (<anonymous>)
    at gotStat (node:fs:2640:19)
    at /<snip>/.pnp.js:9375:13
```
